org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/LinkedHashSetQueue.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.builder.impl;
+
+import java.util.AbstractQueue;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Queue;
+
+/**
+ * {@link Queue} with unique valuess
+ * 
+ * @author Christian Dietrich - Initial contribution and API
+ */
+public class LinkedHashSetQueue<T> extends AbstractQueue<T> {
```

Comment:
Since this impl is not general purpose, I'd like to make this package private.

Reply:
agreeded

Reply:
but our test structure disallows this

Reply:
Understood.

Reply:
Will add our nietend no Instantiate javadoc